pkg/api/types.go

```diff
@@ -1648,6 +1648,20 @@ type SecretList struct {
 	Items []Secret `json:"items"`
 }
 
+type Config struct {
+	TypeMeta   `json:",inline" yaml:",inline"`
+	ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Flags map[string]string `json:"flags,omitempty" yaml:"flags"`
```

Comment:
Absolutely. I'll draft up a proposal asap. Then, we can go over the design and the implementation.